### ISSUE TITLE
Forward declarations are needed for `MIR_output` result to be scanned again

### ISSUE THREAD

**Issue by eliphatfs** [vnmakarov/mir#254]
`MIR_output` generally seems to put proto declarations at the end of modules. When scanning the output again MIR complains about unrecognized symbols. Perhaps forward declarations need to be added in `MIR_output`?

**Comment by Itay2805** [comment]
Yeah, right now I get around that by first going over everything and defining the forwards and only then defining everything, and it's pretty ugly to do.... It would be alot better if mir wouldn't care about order at all...

**Comment by vnmakarov** [comment]
Sorry for slow response.  I am quite busy with some other projects until the end of April.  I'll address this issue later.  It is good that you found a temporary workaround meanwhile.

**Comment by vnmakarov** [comment]
Sorry, I missed your email and thank you for reporting the problem.

I'll think how to resolve the problem.  I could generate forward declaration or prototypes before usages or permitting definitions after the usage.  The later means no sense in forward clauses at all and requires two-pass approach in MIR scanning and probably decreases the speed of scanning.  So I am leaning towards the 1st solution.
